docs/README-Extending.md

```diff
@@ -0,0 +1,133 @@
+# VSCode extensions for Zowe Explorer
+
+The Zowe Explorer provides extension points that assist third party extenders to create extensions that access Explorer entities and enrich the user experience. Extenders should consider there are two mechanisms available by which they extend Zowe Explorer. 
+- VSCode provides extension points via the package.json file relating contextual strings with entity types and containers in the Zowe Explorer and is the way forward with menus for example.
+- Zowe Explorer provides an evolving extenders API that provides specific access to Zowe functionality that for example enables the extender to interact with servers
+using CLI profiles. The intention is to add more such API functions as they are identified as of value. 
+
+## VSCode Extender Package
+
+Extender packages are created as new VSCode extensions. For general information about VSCode extensions, visit the following links:
+
+- [Using extensions in Visual Studio Code](https://code.visualstudio.com/docs/introvideos/extend) 
+- [VSCode Extension API](https://code.visualstudio.com/api) 
+- [vscode-extension-samples](https://github.com/Microsoft/vscode-extension-samples) 
+
+When creating an extension for Zowe Explorer it is necessary to specify that dependency in the package.json file as follows:
+```json	
+    "extensionDependencies": [
+		"zowe.vscode-extension-for-zowe"
+	],
+
+```
+The addition of this script also enforces that the Zowe Explorer is a pre-requisite package on the users machine which will install it if necessary 
+when the extender package is installed. This also ties the activate functionality of both packages enabling direct access to the Zowe Explorer API
+
+## Contextual Hooks
+
+To enable menus, extenders can reference the context values associated with the individual Tree Items. Zowe Explorer uses a strategy of adding and removing 
+elements of context to the individual context values if that imparts additional information that could assist with menu triggers. In the example below we
+are referencing a Job type tree item that has additional information indicated by the *_rc* context. This can be used by an extender to trigger a specifc menu.   
+
+```json
+		"menus": {
+			"view/item/context": [
+				{
+					"when": "view == zowe.jobs && viewItem =~ /^job.*/ && viewItem =~ /^.*_rc=CC.*/",
+					"command": "zowe.testmule.retcode",
+					"group": "4_workspace"
+				}
+            ],
```

Comment:
Does it matter that the menus object is not closed in this documentation?
(I only caught this and wanted to ask because github highlighted text past the end of the json code block as red in the diff)